### PR TITLE
Apply smooth-scroll when `prefers-reduced-motion: no-preference`

### PR DESCRIPTION
**Pull request by ffoodd**
After watching [@mmatuzo's talk "Writing even more CSS with accessibility in mind"](https://www.matuzo.at/blog/writing-even-more-css-with-accessibility-in-mind-progressive-enhancement/), I wanted to suggest optionnal smooth scroll.

I saw #24889 but it's kinda outdated, and with the support of `prefers-reduced-motion` it's more robust now.

Very simple and lightweight PR so feel free to discard it if you think it's still not valuable.

### CODE REVIEW

**Review by patrickhlauke — APPROVED** [review]

**Comment by patrickhlauke** [comment]
i'd be in favour of this

**Review by mdo — APPROVED** [review]
I can dig!

**Comment by mdo** [comment]
We'll want to mention this in the Migration guide, and perhaps add something else in our docs about this.

**Comment by ffoodd** [comment]
Added a mention in our accessibility page, and introduced a reboot section in the migration guide for alpha3.

File: site/content/docs/5.0/getting-started/accessibility.md [hunk 1]
```diff
@@ -47,6 +47,8 @@ For visually hidden interactive controls, such as traditional "skip" links, use
 
 Bootstrap includes support for the [`prefers-reduced-motion` media feature](https://drafts.csswg.org/mediaqueries-5/#prefers-reduced-motion). In browsers/environments that allow the user to specify their preference for reduced motion, most CSS transition effects in Bootstrap (for instance, when a modal dialog is opened or closed, or the sliding animation in carousels) will be disabled.
 
+Alternatively, the `scroll-behavior` property enables smooth scroll if the user did not ask for reduced motion.
```
Comment (patrickhlauke) [inline site/content/docs/5.0/getting-started/accessibility.md:50]: this, in isolation, makes no sense. alternatively to what? i think this needs a bit more of an explanation.

Comment (ffoodd) [reply site/content/docs/5.0/getting-started/accessibility.md:50]: Alternatively to disabling `transitions`s, in my mind :D 
Apart from the beginning of the sentence, using maybe something like "Bootstrap also use…" which would clarify the subject a bit, I'm not quite sure of what to explain more. I'll try something :)

File: scss/_reboot.scss [hunk 2]
```diff
@@ -27,6 +27,10 @@
 
 :root {
   font-size: $font-size-root;
+
+  @media (prefers-reduced-motion: no-preference) {
+    scroll-behavior: smooth;
```
Comment (MartijnCuppens) [inline scss/_reboot.scss:32]: Maybe we should make this configurable (Sass variable)?

File: site/content/docs/5.0/getting-started/accessibility.md [hunk 3]
```diff
@@ -47,7 +47,7 @@ For visually hidden interactive controls, such as traditional "skip" links, use
 
 Bootstrap includes support for the [`prefers-reduced-motion` media feature](https://drafts.csswg.org/mediaqueries-5/#prefers-reduced-motion). In browsers/environments that allow the user to specify their preference for reduced motion, most CSS transition effects in Bootstrap (for instance, when a modal dialog is opened or closed, or the sliding animation in carousels) will be disabled.
 
-Alternatively, the `scroll-behavior` property enables smooth scroll if the user did not ask for reduced motion.
+Bootstrap also enables smooth scroll if the user did not ask for reduced motion, using the `scroll-behavior` property.
```
Comment (patrickhlauke) [inline site/content/docs/5.0/getting-started/accessibility.md:50]: Perhaps a bit verbose, but I'd go for something along the lines of:

```
On browsers that support the `prefers-reduced-motion`, and where the user has *not* explicitly signaled that they'd prefer reduced motion (i.e. where `prefers-reduced-motion: no-preference`), Bootstrap enables smooth scrolling using the `scroll-behavior` property. 
```

File: scss/_reboot.scss (hunk 2 above)
Comment (ffoodd) [reply scss/_reboot.scss]: Neat idea; I introduced `$enable-smooth-scroll` (`true` as a default) to ease override.

File: site/content/docs/5.0/getting-started/accessibility.md (hunk 3 above)
Comment (ffoodd) [reply site/content/docs/5.0/getting-started/accessibility.md]: Very clear, thanks!

**Review by XhmikosR — APPROVED** [review]

**Comment by andrehsu** [comment]
On single-page applications, the effect is really jarring when using client-side navigation, which scrolls to the top of a page after page load. Is there a way to disable this effect? Using
```
html {
    scroll-behavior: auto;
}
```
didn't seem to work.

**Comment by ffoodd** [comment]
@andrehsu Please open an issue instead of reviving a closes pull request.

**Comment by panoply** [comment]
> On single-page applications, the effect is really jarring when using client-side navigation

@andrehsu indeed it is and it should not be the default. 

When using `scroll-behavior: smooth` then`window.scrollTo` will not be respected. It's problematic, especially considering modern web is leveraging client side rendered frameworks. Smooth scroll behaves identical to issues which arise when setting `body, html {  width: 100%; height: 100%; }`. If (for example) you are swiping back and forth between history in web applications that restore scroll positions manually then issues arise. The interesting thing here is that (in chrome at-least) `scroll-behavior: smooth` will work fine when navigating back and forth by buttons, ie: users clicks back/forward button but the moment a user swipes back/forward is where the issues arise and previously referenced scroll positions or scroll positions being applied onload and invoked via `window.scrollTo` will not work. 

The solution or workaround I've found is to explicitly set this to `auto` on the `:root`selector and also asserting an `!important`on the selector. 

```
:root {
  scroll-behavior: auto !important;
}
```

Also, I'm not going to create a new issue for this because reviving an old one is how those come across this defect will be able to better find context regarding the bug, it is also linked in documentation.

**Pull request merged**